### PR TITLE
Added mash modules and nf-tests

### LINKED ISSUE #11: Add mash modules to workflow
### Description of feature

- [x] Install mash/sketch and mash/screen modules using `nf-core install ...`
- [x] Add mash modules to `workflows/phageannotator.nf`
- [x] Use `params.reference_fasta` as input to mash/sketch and `ch_input.fastq` & `MASH_SKETCH.out.mash` as input for mash/screen
- [x] Update `nextflow.config` to include mash parameters
- [x] Update `nextflow_schema.json` to include mash params using `nf-core schema build`
- [x] Update `conf/modules.config` to include mash extra arguments
- [x] Test mash modules using `nextflow run . -profile test,singularity --outdir results`
- [x] Lint additions using `nf-core lint`
- [x] Sync mash branch with nf-core/phageannotater:dev once PR #23  has been merged.
- [x] Submit a PR mash modules

Image

### PR DESCRIPTION
**Pull request by CarsonJM**
## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/phageannotator/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/phageannotator _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/test.config [hunk 1]
```diff
@@ -22,4 +22,6 @@ params {
     // Input data
     input  = 'assets/1.0.0/test_samplesheet.csv'
 
+    // Tool options
+    mash_screen_reference_fasta     = 'https://millardlab-inphared.s3.climb.ac.uk/2Jul2023_refseq_genomes.fa.gz'
```
Comment (sateeshperi) [inline conf/test.config]: will this fasta file be always publicly available? else it will have to move to nf-core/test-datasets

Comment (CarsonJM) [reply conf/test.config]: Great point, I will update this to use an already existing nf-core/test-dataset

Comment (CarsonJM) [inline conf/test.config]:
```suggestion
    mash_screen_reference_fasta     = 'https://github.com/nf-core/test-datasets/blob/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz'
```

**Review by sateeshperi — APPROVED** [review]
gr8 job! 🥇

**Pull request merged**